tfhe/src/core_crypto/gpu/mod.rs

```diff
@@ -1124,6 +1124,24 @@ pub fn synchronize_devices(gpu_count: u32) {
 pub fn check_valid_cuda_malloc(size: u64, gpu_index: GpuIndex) -> bool {
     unsafe { cuda_check_valid_malloc(size, gpu_index.get()) }
 }
+
+/// Check if a memory allocation fits in GPU memory. If it doesn't fit, panic with
+/// a helpful message.
+pub fn check_valid_cuda_malloc_assert_oom(size: u64, gpu_index: GpuIndex) {
```

Comment:
Maybe this function should be `pub(crate)`, I'm not sure users want to panic in this way. :thinking: Wdyt @tmontaigu?